src/inmanta/server/server.py

```diff
@@ -173,7 +175,17 @@ def _ensure_directory_exist(directory: str, *subdirs: str) -> str:
         slices = []
         extension_names = set()
         for slice_name, slice in self._server.get_slices().items():
-            slices.append(SliceStatus(name=slice_name, status=await slice.get_status()))
+            try:
+                slices.append(SliceStatus(name=slice_name, status=await slice.get_status()))
+            except (PostgresError, InterfaceError) as e:
+                LOGGER.warning(
+                    f"The following error occured with the database,"
+                    f" while trying to determine the status of slice {slice_name}",
+                    e,
+                    exc_info=True,
+                )
+                raise ServiceUnavailable(str(e))
```

Comment:
Can you add that to the docstring of the method? So that the swagger gets updated as well.

We should only catch a generic exception here. The database unavailable should be handled in the get status of the database slice.